src/SFA.DAS.ApplyService.Web/Infrastructure/FileValidator.cs

```diff
@@ -26,27 +26,65 @@ public static bool FileValidationPassed(List<Answer> answers, Page page, List<Va
 
                     var fileNameParts = file.FileName.Split(".", StringSplitOptions.RemoveEmptyEntries);
                     var fileNameExtension = fileNameParts[fileNameParts.Length - 1];
-                    
+
                     if (fileNameExtension != allowedExtension || file.ContentType.ToLower() != mimeType)
                     {
                         modelState.AddModelError(file.Name, typeValidation.ErrorMessage);
                         errorMessages.Add(new ValidationErrorDetail(file.Name, typeValidation.ErrorMessage));
                         fileValidationPassed = false;
                     }
+                    else if (!FileContentIsValidForMimeType(fileNameExtension, file.OpenReadStream()))
+                    {
+                        var errorMessage = "File content is incorrect for the specified file type";
+                        modelState.AddModelError(file.Name, errorMessage);
+                        errorMessages.Add(new ValidationErrorDetail(file.Name, errorMessage));
+                        fileValidationPassed = false;
+                    }
                     else
                     {
                         // Only add to answers if type validation passes.
-                        answers.Add(new Answer() {QuestionId = file.Name, Value = file.FileName});
+                        answers.Add(new Answer() { QuestionId = file.Name, Value = file.FileName });
                     }
                 }
                 else
                 {
                     // Only add to answers if type validation passes.
-                    answers.Add(new Answer() {QuestionId = file.Name, Value = file.FileName});
+                    answers.Add(new Answer() { QuestionId = file.Name, Value = file.FileName });
                 }
             }
 
             return fileValidationPassed;
         }
+
+        // Refer to this: https://www.garykessler.net/library/file_sigs.html
+        private static Dictionary<string, byte[]> _knownHeaders = new Dictionary<string, byte[]>
+        {
+            {"JPG",  new byte[] { 0xFF, 0xD8, 0xFF }},
+            {"JPEG", new byte[] { 0xFF, 0xD8, 0xFF }},
+            {"TIF",  new byte[] { 0x49, 0x49, 0x2A, 0x00 }},
+            {"TIFF", new byte[] { 0x49, 0x49, 0x2A, 0x00 }},
+            {"GIF",  new byte[] { 0x47, 0x49, 0x46, 0x38 }},
+            {"BMP",  new byte[] { 0x42, 0x4D }},
+            {"ICO",  new byte[] { 0x00, 0x00, 0x01, 0x00 }},
+            {"PDF",  new byte[] { 0x25, 0x50, 0x44, 0x46 }}
+        };
```

Comment:
I like this, but why are we now taking responsibility to make sure a file is actually a type of that file?  If the user uploads a .pdf, but it's not really a pdf, then that's their problem surely and they shouldn't be renaming extensions etc. Making this check and showing an error then pushes the problem back to us as we have to then explain why the file isn't uploading.